backend/src/main/java/com/bakdata/conquery/models/auth/web/DefaultAuthFilter.java

```diff
@@ -0,0 +1,112 @@
+package com.bakdata.conquery.models.auth.web;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.ws.rs.NotAuthorizedException;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.PreMatching;
+import javax.ws.rs.core.SecurityContext;
+
+import com.bakdata.conquery.models.auth.AuthorizationController;
+import com.bakdata.conquery.models.auth.ConqueryAuthenticationRealm;
+import com.bakdata.conquery.models.auth.ConqueryAuthenticator;
+import com.bakdata.conquery.models.auth.entities.User;
+import com.google.common.base.Preconditions;
+import io.dropwizard.auth.AuthFilter;
+import io.dropwizard.auth.DefaultUnauthorizedHandler;
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import lombok.experimental.Accessors;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.shiro.authc.AuthenticationToken;
+import org.apache.shiro.realm.Realm;
+
+/**
+ * This filter hooks into dropwizard's request handling to extract and process
+ * security relevant information for protected resources. The request is first
+ * submitted to the registered {@link ConqueryAuthenticationRealm}s for the
+ * token extraction, then the extracted tokens are submitted these realms
+ * through Dropwizards {@link AuthFilter} and Shiro.
+ */
+@Slf4j
+@PreMatching
+@RequiredArgsConstructor(access = AccessLevel.PRIVATE)
+public class DefaultAuthFilter extends AuthFilter<AuthenticationToken, User> {
+
+	private final AuthorizationController controller;
+
+	@Override
+	public void filter(final ContainerRequestContext requestContext) throws IOException {
+
+		// The token extraction process
+		List<AuthenticationToken> tokens = new ArrayList<>();
+		for (ConqueryAuthenticationRealm realm : controller.getAuthenticationRealms()) {
+			AuthenticationToken token = null;
+			if ((token = realm.extractToken(requestContext)) != null) {
+				log.trace("Realm {} extracted a token form the request: {}", ((Realm) realm).getName(), token);
+				tokens.add(token);
+			}
+			log.trace("Realm {} did not extract a token form the request.", ((Realm) realm).getName());
+		}
+
+		if (tokens.isEmpty()) {
+			log.warn("No tokens could be parsed from the request");
+			throw new NotAuthorizedException("Failed to authenticate request. The cause has been logged.");
+		}
+
+		List<AuthenticationToken> failedAuthentications = new ArrayList<>();
+
+		// The authentication process
+		for (AuthenticationToken token : tokens) {
+			try {
+				// Submit the token to dropwizard which forwards it to Shiro
+				if (!authenticate(requestContext, token, SecurityContext.BASIC_AUTH)) {
+					throw new NotAuthorizedException("Authentication failed", "Bearer");
```

Comment:
Musst du hier nicht eher ein Continue machen? Sonst authentifizierst du ja nur das erste Token

Reply:
Die for-Schleife über die Token ist um das try-catch. Generell sollte bei einem validen Token aber keine Exception geworfen werden.